FunKiiU.py

```diff
@@ -370,15 +386,60 @@ def main(titles=None, keys=None, onlinekeys=False, onlinetickets=False, download
         for title_data in titlekeys_data:
             title_id = title_data['titleID']
             title_key = title_data.get('titleKey', None)
-            name = title_data.get('name', None)
             region = title_data.get('region', None)
+            name = title_data.get('name', None)
             typecheck = title_id[4:8]
 
             # skip system stuff (try to only get games+updates+dlcs)
             if typecheck in ('8005', '800f') or int(typecheck, 16) & 0x10:
                 continue
             elif title_id in titles:
                 continue
+            process_title_id(title_id, title_key, region, name, output_dir, retry_count, onlinetickets, patch_demo, patch_dlc)
+
+    if download_eur:
+        for title_data in titlekeys_data:
+            title_id = title_data['titleID']
+            title_key = title_data.get('titleKey', None)
+            region = title_data.get('region', None)
+            name = title_data.get('name', None)
+            typecheck = title_id[4:8]
+
+            # skip system stuff (try to only get games+updates+dlcs)
+            if typecheck in ('8005', '800f') or int(typecheck, 16) & 0x10 or region in ('USA','JPN'):
```

Comment:
If you have three times exactly the same block, that only differ in one part, make that part a variable and set it beforehand, then you end up with only one block.
Don't copy paste :-)